ta/sks/src/entry.c

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018-2019, Linaro Limited
+ */
+
+#include <compiler.h>
+#include <sks_ta.h>
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+
+TEE_Result TA_CreateEntryPoint(void)
+{
+	return TEE_SUCCESS;
+}
+
+void TA_DestroyEntryPoint(void)
+{
+}
+
+TEE_Result TA_OpenSessionEntryPoint(uint32_t __unused param_types,
+				    TEE_Param __unused params[4],
+				    void **session)
+{
+	*session = NULL;
+
+	return TEE_SUCCESS;
+}
+
+void TA_CloseSessionEntryPoint(void *session __unused)
+{
+}
+
+/*
+ * Entry point for invocation command SKS_CMD_PING
+ *
+ * @ctrl - param memref[0] or NULL: expected NULL
+ * @in - param memref[1] or NULL: expected NULL
+ * @out - param memref[2] or NULL
+ *
+ * Return a TEE_Result value
+ */
+static TEE_Result entry_ping(TEE_Param *ctrl, TEE_Param *in, TEE_Param *out)
+{
+	const uint32_t ver[] = { SKS_VERSION_MAJOR, SKS_VERSION_MINOR };
+
+	if (ctrl || in)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (!out)
+		return TEE_SUCCESS;
+
+	if (out->memref.size < sizeof(ver)) {
+		out->memref.size = sizeof(ver);
+		return TEE_ERROR_SHORT_BUFFER;
+	}
+
+	TEE_MemMove(out->memref.buffer, ver, sizeof(ver));
+
```

Comment:
`out->memref.size = sizeof(ver);` to signal amount of returned data?

Reply:
right, thanks.